examples/api-wrapper/src/lib.rs

```diff
@@ -0,0 +1,176 @@
+/*
+Example app to demonstrate how to wrap several API calls into single edge app.
+This example gets device status from SmartThings API and toggles it
+App needs following env vars to be set:
+PASSWORD - password to check user's permissions, simplest form of authentication
+DEVICE - device ID in SmartThings
+TOKEN - SmartThings API token
+*/
+
+use std::env;
+use fastedge::{
+    body::Body,
+    http::{header, Error, Method, Request, Response, StatusCode},
+};
+use url::Url;
+
+const API_BASE: &str = "https://api.smartthings.com/v1/devices/";
+
+#[fastedge::http]
+fn main(req: Request<Body>) -> Result<Response<Body>, Error> {
+    match req.method() {
+        &Method::GET | &Method::HEAD => (),
+        _ => return Response::builder().status(StatusCode::METHOD_NOT_ALLOWED).header(header::ALLOW, "GET, HEAD").body(Body::from("This method is not allowed\n"))
+    };
+
+    let expected_pass = match env::var("PASSWORD") {
+        Err(_) => return Response::builder().status(StatusCode::INTERNAL_SERVER_ERROR).body(Body::from("Misconfigured app\n")),
+        Ok(r) => r
+    };
```

Comment:
This kind of expression can be written in a simpler way:
```
let Ok(expected_pass) = env::var("PASSWORD") else {
    return Response::builder().status(StatusCode::INTERNAL_SERVER_ERROR).body(Body::from("Misconfigured app\n"));
}
```

Reply:
changed